app/models/financial_transaction.rb

```diff
@@ -38,12 +43,9 @@ def self.ransackable_associations(_auth_object = nil)
     %w[] # none, and certainly not user until we've secured that more
   end
 
-  # Use this save method instead of simple save and after callback
-  def add_transaction!
-    ordergroup.add_financial_transaction! amount, note, user, financial_transaction_type
-  end
-
   def revert!(user)
+    raise 'Pending Transaction cannot be reverted' if amount.nil?
+
```

Comment:
Not sure how good the testing of reverting transactions is, ideally we'd test this.
But as we also want to get this PR merged, feel free to skip if this feature has no testing at all.